File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> frequency;
        //create a count map
        for (int num : nums) {
            frequency[num] += 1;
        }
        //do build heap
        priority_queue<pair<int, int>> maxHeap;
        for (auto const& [key, value]: frequency) {
            //sort by size
            maxHeap.push({value, key});
        }

        vector<int> result;

        for (int i = 0; i < k; i++) {
            result.push_back(maxHeap.top().second);
            maxHeap.pop();
        }
        return result;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashSet = {}
        for i in range(len(nums)):
            num = nums[i]
            if num in hashSet:
                return [hashSet[num], i]
            hashSet[target - num] = i
        return False